### DISCUSSION TITLE
warning: could not open directory 'Application Data/': Permission denied

### DISCUSSION THREAD

**Discussion by anurag-kkumar** [community/community#60015] [Programming Help]
### Body

it found after $ git add "my file name "
                     $ git status
                      warning: could not open directory 'Application Data/': Permission
denied
warning: could not open directory 'Cookies/': Permission denied
warning: could not open directory 'Local Settings/': Permission de
nied
warning: could not open directory 'My Documents/': Permission deni
ed
warning: could not open directory 'NetHood/': Permission denied
warning: could not open directory 'PrintHood/': Permission denied
warning: could not open directory 'Recent/': Permission denied
warning: could not open directory 'SendTo/': Permission denied
warning: could not open directory 'Start Menu/': Permission denied
warning: could not open directory 'Templates/': Permission denied
On branch Firstclass.java

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        -1.14-windows.xml
        .BigNox/
        .android/
        .bash_history
        .cache/
        .gitconfig
        .lesshst
        .redhat/
        .vscode/
        3D Objects/
        ABCD/
        ANURAG.docx
        AppData/
        Contacts/
        Desktop/
        Documents/
        Downloads/
        Favorites/
        IdeaProjects/
        IntelGraphicsProfiles/
        Links/
        MicrosoftEdgeBackups/
        Music/
        NTUSER.DAT
        NTUSER.DAT{c4525b16-510f-11eb-a80b-f3b0b19ac261}.TM.blf
        NTUSER.DAT{c4525b16-510f-11eb-a80b-f3b0b19ac261}.TMContain
er00000000000000000001.regtrans-ms
        NTUSER.DAT{c4525b16-510f-11eb-a80b-f3b0b19ac261}.TMContain
er00000000000000000002.regtrans-ms
        Nox_share/
        OneDrive/
        Pictures/
        Saved Games/
        Searches/
        Videos/
        d4ac4633ebd6440fa397b84f1bc94a3c.7z
        ex-list
        firstclass.java
        firstclass1.java
        inittk.ini
        inst.ini
        letslerngit/
        ntuser.dat.LOG1
        ntuser.dat.LOG2
        ntuser.ini
        nuuid.ini
        useruid.ini
        vmlogs/

nothing added to commit but untracked files present (use "git add"
 to track)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by nirav-gajera** [comment]
Create a .gitignore file in the root of your repository using a text editor or the terminal.

Add the paths to the directories you want to exclude from version control. For example:

`# .gitignore
Application Data/
Cookies/
Local Settings/
My Documents/
NetHood/
PrintHood/
Recent/
SendTo/
Start Menu/
Templates/
`

Save the .gitignore file.

**Comment by davea42** [comment]
The documentation  I have (so far) found on .gitignore
is remarkably free of names with spaces. 
Anyway, in .gitignore each single pattern should be on
its own line.  

git-scm.com/docs/gitignore
is useful.

To restrict a pattern to just match a directory name
add a trailing / as nirav-gajera suggests.  Without
the trailing / (slash)  gitignore will ignore
file and  directory names.

Having spaces in directory or file names
is no problem, just type it into .gitignore
as it is spelled.

It is very easy to experiment to determine how .gitignore works,
so experiment!

Hope this helps.
